src/common/Backend.ts

```diff
@@ -133,6 +133,31 @@ export class ServerError extends Error {
         return new ServerError(json.message, json.description, json.status);
     }
 }
+export class CancelError extends Error {
+    type = 'cancel';
+
+    message: string;
+
+    config: AxiosRequestConfig | undefined;
+
+    constructor(message: string | undefined, config?: AxiosRequestConfig) {
+        super();
+        this.message = message ?? 'The request was cancelled';
+        this.config = config;
+    }
+
+    static isCancel(cancel: unknown): cancel is CancelError {
+        if (typeof cancel !== 'object' || cancel === null) {
+            return false;
+        }
+        const obj = cancel as Record<string, unknown>;
+        return typeof obj.message === 'string' && obj.type === 'cancel';
+    }
```

Comment:
Why are you doing all that when `x instanceof CancelError` exists?

Reply:
I was just doing what we did for server error